Core/Src/Systems/External/Esc.c

```diff
@@ -1,16 +1,19 @@
-#include "../../../Inc/Sensors/CANSensors/Esc.h"
+#include "../../../Inc/Systems/External/Esc.h"
 
 #include <stdio.h>  // For printf
 
 void initEsc(Esc* esc, int hz) {
-    if (esc == NULL) {
-        return;
+    if (esc != NULL) {
+        // initCANSensor(&esc->base, "Esc", hz, 0);
+        esc->temp = 0.0;
+        esc->motor_revolutions = 0.0;
+        // esc->base.sensor.updateable.update = updateEsc;
     }
 
-    initCANSensor(&esc->base, "Esc", hz);
+    // initCANSensor(&esc->base, "Esc", hz);
     esc->temp = 0.0;
     esc->motor_revolutions = 0.0;
-    esc->base.sensor.updateable.update = updateEsc;
+    // esc->base.sensor.updateable.update = updateEsc;
 }
```

Comment:
The code performs initialization both inside and outside the NULL check, which could lead to crashes. 
For example, ```esc->temp = 0.0;``` from outside the if statement could crash if ```esc``` is ```NULL```, since ```esc``` would be a null pointer that points to an invalid memory location.

Example fix with early return pattern:
```c
void initEsc(Esc* esc, int hz) {
    if (esc == NULL) {
        return;
    }
    esc->temp = 0.0;
    esc->motor_revolutions = 0.0;
}
```

Reply:
Should this return with an error/success?